docs/developer/security-review.md

```
@@ -303,3 +307,34 @@ lock (#204), removal of unpinned `@playwright/mcp@latest` (#205). Ledger refresh
+
+### 2026-08-31 — R4 temp-file hardening (round 4)
+
+**Scope.** Predictable temp path `printf '/tmp/fwlive-nft-%s' "$$"` fallback in `build_rules_map` (3 sites) gave a root-level arbitrary-write primitive via symlink at `/tmp/fwlive-*` (PIDs brute-forceable). `mktemp` present on OpenWrt (BusyBox) so fallback unlikely but must not exist in root-context code. Same class as #204 (lock symlink).
+
+**Method.** Grep for `$$` / `printf.*fwlive`; host reproduction under `dash` + `busybox sh` with `mktemp` shadowed (absent).
+
+**Fix (minimal, POSIX sh).**
+
+- **Delete fallback:** removed `|| printf '/tmp/fwlive-nft-%s' "$$"` from all 3 sites (no predictable-path code path remains).
+- **Single helper:** `_fwlive_mktemp <prefix>` — `mktemp "/tmp/<prefix>.XXXXXX" 2>/dev/null` only. `TMPDIR` is **deliberately not honoured** and there is **no bare-`mktemp` fallback**: bare `mktemp` consults `TMPDIR`, and an attacker-writable non-sticky `TMPDIR` would restore the create-then-reopen (TOCTOU) symlink-write primitive, since `build_rules_map` reopens the path with `>` while running as root. A fixed sticky `/tmp` removes the vector. Never `rm`+reuse, never `touch`/`chmod`. If `mktemp` fails, the backend enrichment is skipped and a well-formed map is still returned. BusyBox `mktemp` sane.
+- **Graceful degradation:** ` _tmp=$(_fwlive_mktemp fwlive-nft) || _tmp=''` then `if [ -n "$_tmp" ]; then nft_list_ruleset >"$_tmp" ...; rm -f "$_tmp"; fi` — if `mktemp` absent/failing, backend enrichment skipped, still returns well-formed `{"backend":...,"rules":{...}}` with UCI names. No fixed-path write.
```

markdownlint-cli2 (0.23.2)

[warning] Spaces inside code span elements

(MD038, no-space-in-code)